exercises/lamar_moore/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,149 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <algorithm>
+#include <vector>
+#include <map>
+
+using namespace std;
+
+namespace
+{
+	/** Removes the double dash character "--" from strings
+
+	@param str String which might contain "--"
+	*/
+	vector<string> splitDoubleDash(string &str)
+	{
+		vector<string> stringList;
+		size_t pos;
+
+		if((pos = str.find("--", 0)) != string::npos)
+		{
+			stringList.push_back(string(str.begin(), str.begin()+pos));
+			stringList.push_back(string(str.begin()+pos+2, str.end()));
+		}
+		else
+			stringList.push_back(str);
+
+		return stringList;
+	}
+
+	/** Predicate function which tests whether or not character is a punctuation character
+
+	@param c character tested against the list ".,?'"!():"
+	*/
+	bool isPunctuation(char c)
+	{
+		bool isPunc = false;
+
+		static const char chars[] = ".,?\'\"!():";
+
+		for(int i=0; i<strlen(chars); i++)
```

Comment:
Failing RHEL6 build. strlen is defined. As the  I think you need to include <cstring> to use this, or just convert your char * to as std::string and use std::string::size()